apps/server/src/shared/domain/rules/submission.rule.spec.ts

```diff
@@ -0,0 +1,194 @@
+import { MikroORM } from '@mikro-orm/core';
+import { Test, TestingModule } from '@nestjs/testing';
+import {
+	courseFactory,
+	courseGroupFactory,
+	roleFactory,
+	setupEntities,
+	submissionFactory,
+	taskFactory,
+	userFactory,
+} from '@shared/testing';
+import { CourseGroupRule, CourseRule, LessonRule, SubmissionRule, TaskRule } from '.';
+import { User } from '../entity';
+import { Permission, RoleName } from '../interface';
+import { Actions } from './actions.enum';
+
+describe('SubmissionRule', () => {
+	let orm: MikroORM;
+	let submissionRule: SubmissionRule;
+	let student: User;
+	let student2: User;
+	let teacher: User;
+	const permissionA = 'a' as Permission;
+	const permissionB = 'b' as Permission;
+	const permissionC = 'c' as Permission;
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [TaskRule, CourseRule, LessonRule, CourseGroupRule, SubmissionRule],
+		}).compile();
+
+		submissionRule = await module.get(SubmissionRule);
+
+		const studentRole = roleFactory.build({ permissions: [permissionA, permissionB], name: RoleName.STUDENT });
```

Comment:
We start to avoid data preparations in beforeEach / beforeAll ..can you please look into the files-storage test how the setup methods look and the helper functions for recreating of same data structures. 
In short all mocks and all entitys for one senerio are in one setup method. If the setup for one test different we create a new senerio with different setup.

Reply:
I refactored it that way (though I think it is not good).